File: benchmarks/vm/17_string_call_loop.lua
local function pick_label(i)
    local r = i % 4
    if r == 0 then return "alpha" end
    if r == 1 then return "beta" end
    if r == 2 then return "gamma" end
    return "delta"
end

local function label_score(label)
    if label == "alpha" then return 11 end
    if label == "beta" then return 17 end
    if label == "gamma" then return 23 end
    return 31
end

local function nested_score(i)
    return label_score(pick_label(i))
end

local t0 = os.clock()
local total = 0
for i = 0, 119999 do
    total = total + nested_score(i)
end
local total_ms = (os.clock() - t0) * 1000
print("total = " .. tostring(total))
print(string.format("BENCH_RESULT: %.3f ms", total_ms))

File: benchmarks/vm/17_string_call_loop.rb
def pick_label(i)
  case i % 4
  when 0 then "alpha"
  when 1 then "beta"
  when 2 then "gamma"
  else "delta"
  end
end

def label_score(label)
  case label
  when "alpha" then 11
  when "beta" then 17
  when "gamma" then 23
  else 31
  end
end

def nested_score(i)
  label_score(pick_label(i))
end

t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
total = 0
120_000.times do |i|
  total += nested_score(i)
end
total_ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t0) * 1000
puts "total = #{total}"
puts format("BENCH_RESULT: %.3f ms", total_ms)
